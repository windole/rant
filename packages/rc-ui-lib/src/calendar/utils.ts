/* eslint-disable no-nested-ternary */
export function compareMonth(date1: Date, date2: Date): number {
  const year1 = date1.getFullYear();
  const year2 = date2.getFullYear();

  if (year1 === year2) {
    const month1 = date1.getMonth();
    const month2 = date2.getMonth();
    return month1 === month2 ? 0 : month1 > month2 ? 1 : -1;
  }

  return year1 > year2 ? 1 : -1;
}

export function compareDay(day1: Date, day2: Date): number {
  const compareMonthResult = compareMonth(day1, day2);

  if (compareMonthResult === 0) {
    const date1 = day1.getDate();
    const date2 = day2.getDate();
    return date1 === date2 ? 0 : date1 > date2 ? 1 : -1;
  }

  return compareMonthResult;
}

export const cloneDate = (date: Date): Date => new Date(date);

export const cloneDates = (dates: Date | Date[]): Date | Date[] =>
  Array.isArray(dates) ? dates.map(cloneDate) : cloneDate(dates);

export function getDayByOffset(date: Date, offset: number): Date {
  const cloned = cloneDate(date);
  cloned.setDate(cloned.getDate() + offset);
  return cloned;
}

export const getPrevDay = (date: Date): Date => getDayByOffset(date, -1);
export const getNextDay = (date: Date): Date => getDayByOffset(date, 1);
export const getToday = (): Date => {
  const today = new Date();
  today.setHours(0, 0, 0, 0);
  return today;
};

export function calcDateNum(date: [Date, Date]): number {
  const day1 = date[0].getTime();
  const day2 = date[1].getTime();
  return (day2 - day1) / (1000 * 60 * 60 * 24) + 1;
}

export const getMonthEndDay = (year: number, month: number): number =>
  32 - new Date(year, month - 1, 32).getDate();

const calendarLang = {
  end: '结束',
  start: '开始',
  title: '日期选择',
  confirm: '确定',
  startEnd: '开始/结束',
  weekdays: ['日', '一', '二', '三', '四', '五', '六'],
  monthTitle: (year: number, month: number) => `${year}年${month}月`,
  rangePrompt: (maxRange: number) => `最多选择 ${maxRange} 天`,
};

export const t = (key: string, ...args): string => {
  if (args.length) {
    return calendarLang[key](...args);
  }
  return calendarLang[key];
};

export const formatMonthTitle = (date: Date): string =>
  t('monthTitle', date.getFullYear(), date.getMonth() + 1);
