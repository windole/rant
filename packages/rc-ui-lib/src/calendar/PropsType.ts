import React, { MouseEvent } from 'react';
import { PopupPosition } from '../popup';
import { BaseTypeProps, TeleportType } from '../utils';

export type CalendarType = 'single' | 'range' | 'multiple';

export type CalendarDayType =
  | ''
  | 'start'
  | 'start-end'
  | 'middle'
  | 'end'
  | 'selected'
  | 'multiple-middle'
  | 'multiple-selected'
  | 'disabled'
  | 'placeholder';

export type CalendarDayItem = {
  date?: Date;
  text?: string | number;
  type?: CalendarDayType;
  topInfo?: string;
  className?: string;
  bottomInfo?: string;
};

export type CalendarDayProps = {
  item: CalendarDayItem;
  color: string;
  index: number;
  offset?: number;
  rowHeight: string;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClick?: (item: CalendarDayItem) => void;
};

export type CalendarInstance = {
  reset: (date?: Date | Date[]) => void;
  scrollToDate: (targetDate: Date) => void;
};

export interface CalendarProps extends BaseTypeProps {
  position?: PopupPosition;
  round?: boolean;
  closeOnPopstate?: boolean;
  closeOnClickOverlay?: boolean;
  safeAreaInsetBottom?: boolean;
  teleport?: TeleportType;
  show?: boolean;
  type?: CalendarType;
  title?: React.ReactNode | string;
  subtitle?: React.ReactNode | string;
  color?: string;
  readonly?: boolean;
  poppable?: boolean;
  maxRange?: string | number;
  showMark?: boolean;
  showTitle?: boolean;
  formatter?: (item: CalendarDayItem) => CalendarDayItem;
  rowHeight?: string | number;
  confirmText?: string;
  rangePrompt?: string;
  lazyRender?: boolean;
  showConfirm?: boolean;
  defaultDate?: Date | Date[] | null;
  allowSameDay?: boolean;
  showSubtitle?: boolean;
  showRangePrompt?: boolean;
  confirmDisabledText?: string;
  minDate?: Date;
  maxDate?: Date;
  firstDayOfWeek?: number;
  footer?: React.ReactNode | string;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClickSubtitle?: (ev: MouseEvent) => void;
  /** 点击确认后的回调 */
  onConfirm?: (v: Date | Date[]) => void;
  onOverRange?: () => void;
  onSelect?: (v: Date | Date[]) => void;
  onUnselect?: (v: Date | Date[]) => void;
  onMonthShow?: (obj: { date: Date; title: string }) => void;
  onClose?: () => void;
}

export type CalendarMonthProps = {
  date?: Date;
  type?: CalendarType;
  color?: string;
  minDate?: Date;
  maxDate?: Date;
  showMark?: boolean;
  rowHeight?: number | string;
  formatter?: (item: CalendarDayItem) => CalendarDayItem;
  formatMonthTitle?: (date: Date) => React.ReactNode;
  lazyRender?: boolean;
  currentDate?: Date | Date[];
  allowSameDay?: boolean;
  showSubtitle?: boolean;
  showMonthTitle?: boolean;
  firstDayOfWeek?: number;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClick?: (item: CalendarDayItem) => void;
};

export type CalendarMonthInstance = {
  showed?: boolean;
  getTitle: () => string | React.ReactNode;
  getHeight: () => number;
  setVisible: (value?: boolean | undefined) => void;
  scrollToDate: (body: Element, targetDate: Date) => void;
};
