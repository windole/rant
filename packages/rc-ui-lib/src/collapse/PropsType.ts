import { ReactNode } from 'react';
import { CellProps } from '../cell/PropsType';
import { BaseTypeProps } from '../utils';

export interface CollapseProps extends BaseTypeProps {
  /** 是否开启手风琴模式	 */
  accordion?: boolean;
  /** 默认打开的面板name */
  initValue?: string | number | Array<number | string>;
  /** 是否显示外边框	 */
  border?: boolean;
  /** 当前展开面板的 name	 */
  value?: number | string | Array<number | string>;
  /** 子组件	 */
  children?: ReactNode;
  /** 切换面板时触发	*/
  onChange?: (activeNames: number | string | Array<number | string>) => void;
}

export interface CollapseItemProps extends CellProps {
  /** 唯一标识符，默认为索引值	 */
  name?: number | string;
  /** 是否禁用面板	 */
  disabled?: boolean;
  /** 是否为只读状态，只读状态下无法操作面板	 */
  readonly?: boolean;
  /** @private */
  index?: number;
  /** 子组件	 */
  children?: ReactNode;
  // onClick: () => void;
}

export type CollapseItemInstance = {
  toggle: (expand: boolean) => void;
};
