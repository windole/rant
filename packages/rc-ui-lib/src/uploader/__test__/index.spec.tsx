import React, { useRef } from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import Uploader, { UploaderInstance } from '..';
import Icon from '../../icon';
import TestsDOM from '../../../tests/dom';
import { UploaderFileListItem, UploaderProps } from '../PropsType';
import { sleep } from '../../../tests/utils';

describe('<Uploader/>', () => {
  const mockFile = new File([new ArrayBuffer(10000)], 'foo.png', {
    type: 'image/png',
  });

  const mockTextFile = new File([new ArrayBuffer(10000)], 'foo.txt', {
    type: 'text/plain',
  });

  const mockFileDataUrl = 'data:image/test';

  function mockReadAsText(this: FileReader) {
    if (this.onload) {
      this.onload({
        target: {
          result: mockFileDataUrl,
        },
      } as ProgressEvent<FileReader>);
    }
  }

  Object.defineProperty(window.FileReader.prototype, 'readAsText', {
    value: mockReadAsText,
  });

  Object.defineProperty(window.FileReader.prototype, 'readAsDataURL', {
    value: mockReadAsText,
  });

  function createUploader(props: UploaderProps) {
    const { container, rerender } = render(<Uploader {...props} />);

    return {
      container,
      input: () => TestsDOM.mustQuerySelector<HTMLInputElement>(container, '.rc-uploader__input'),
      rerender,
    };
  }

  it('disabled', async () => {
    const onChange = jest.fn();

    const { input } = createUploader({
      disabled: true,
      onChange,
    });

    fireEvent.change(input(), { target: { files: [mockFile] } });

    await waitFor(() => expect(onChange).toBeCalledTimes(0));
  });

  it('read file', async () => {
    const onChange = jest.fn();

    const { input } = createUploader({
      onChange: ([file]) => {
        expect(file.content).toBe(mockFileDataUrl);
        onChange();
      },
    });

    fireEvent.change(input(), { target: { files: [mockFile] } });

    await waitFor(() => expect(onChange).toBeCalledTimes(1));
  });

  it('beforeRead', async () => {
    const afterRead = jest.fn();

    const { input } = createUploader({
      beforeRead: async (files: File[]) => {
        return new Promise<File[]>((resolve) => {
          resolve(files);
        });
      },
      afterRead,
    });

    fireEvent.change(input(), { target: { files: [mockFile] } });

    await waitFor(() => expect(afterRead).toHaveBeenCalledTimes(1));
  });

  it('thumbnail', async () => {
    const fileUrl = 'https://upload.com/foo.txt';
    const imageUrl = 'https://dummyimage.com/300x200/fff.png';

    const { container } = createUploader({
      value: [
        {
          status: 'uploading',
          content: imageUrl,
        },
        {
          status: 'failed',
          content: imageUrl,
          message: 'failed',
        },
        {
          status: 'done',
          content: fileUrl,
          message: 'done',
        },
        {
          status: 'done',
          content: fileUrl,
          removable: false,
          message: 'done',
        },
      ],
    });

    expect(container).toMatchSnapshot();
  });

  it('max-count', async () => {
    const onChange = jest.fn();

    const props: UploaderProps = {
      maxCount: 2,
      multiple: true,
      value: [
        {
          url: 'https://dummyimage.com/600x400/000/fff.png',
          status: 'uploading',
          thumbnail: false,
        },
      ],
      onChange: (files) => {
        onChange(files.length);
      },
    };

    const { container, input, rerender } = createUploader(props);

    fireEvent.change(input(), {
      target: { files: [mockFile, mockFile, mockFile] },
    });

    await waitFor(() => expect(onChange).toBeCalledWith(2));

    const props2 = {
      ...props,
      value: [
        {
          isImage: false,
          url: 'https://upload.com/foo.txt',
          status: 'done',
          thumbnail: false,
        },
        {
          isImage: true,
          url: 'https://dummyimage.com/300x200/fff.png',
          status: 'done',
          content: 'https://dummyimage.com/50/fff.png',
        },
      ],
    };
    rerender(<Uploader {...props2} />);

    expect(container).toMatchSnapshot();
  });

  it('render children', () => {
    const { container } = createUploader({
      previewImage: false,
      value: [],
      children: <div>children</div>,
    });

    expect(container).toMatchSnapshot();
  });

  it('test preview', async () => {
    const onPreview = jest.fn();
    const onClosePreview = jest.fn();

    const { container } = createUploader({
      value: [
        {
          isImage: false,
          url: 'https://upload.com/foo.txt',
          status: 'done',
          thumbnail: false,
          file: mockTextFile,
        },
        {
          isImage: true,
          url: 'https://dummyimage.com/300x200/fff.png',
          status: 'done',
          content: 'https://dummyimage.com/50/fff.png',
        },
      ],
      onClickPreview: (file: UploaderFileListItem) => {
        onPreview(file.url);
      },
      onClosePreview,
      previewImage: true,
      previewFullImage: true,
    });

    const file = TestsDOM.mustQuerySelector(container, '.rc-uploader__file');

    fireEvent.click(file);

    expect(onPreview).toBeCalledWith('https://upload.com/foo.txt');

    const image = TestsDOM.mustQuerySelector(container, '.rc-uploader__preview-image');

    fireEvent.click(image);

    expect(onPreview).toBeCalledWith('https://dummyimage.com/300x200/fff.png');

    const imagePreview = TestsDOM.mustQuerySelector(document.body, '.rc-image-preview__image');

    fireEvent.click(imagePreview);
    expect(onClosePreview).toHaveBeenCalled();
  });

  it('test remove', async () => {
    const onDelete = jest.fn().mockResolvedValue(false);

    const { container } = createUploader({
      value: [
        {
          isImage: false,
          url: 'https://upload.com/foo.txt',
          status: 'done',
          thumbnail: false,
        },
        {
          isImage: true,
          url: 'https://dummyimage.com/300x200/fff.png',
          status: 'done',
          content: 'https://dummyimage.com/50/fff.png',
        },
      ],
      onDelete,
    });

    expect(container).toMatchSnapshot();

    const target = TestsDOM.mustQuerySelector(container, '.rc-uploader__preview-delete');

    fireEvent.click(target);
    await waitFor(() => expect(onDelete).toBeCalledTimes(1));
    expect(container).toMatchSnapshot();
  });

  it('test previewCoverRender', async () => {
    const { container } = createUploader({
      value: [
        {
          // isImage: false,
          url: 'https://upload.com/foo.txt',
          status: 'done',
          thumbnail: false,
        },
        {
          isImage: true,
          url: 'https://dummyimage.com/300x200/fff.png',
          status: 'done',
          content: 'https://dummyimage.com/50/fff.png',
        },
      ],
      previewCoverRender: (item: UploaderFileListItem) => <p>{item.url}</p>,
    });

    expect(container).toMatchSnapshot();
  });

  it('test before read return false', async () => {
    const afterRead = jest.fn();

    const props: UploaderProps = {
      beforeRead: () => undefined,
      afterRead,
    };

    const { input } = createUploader(props);

    fireEvent.change(input(), { target: { files: [mockFile] } });

    expect(afterRead).toHaveBeenCalledTimes(0);
  });

  it('before read return promise and resolve no value', async () => {
    const afterRead = jest.fn();

    const props: UploaderProps = {
      beforeRead: () =>
        new Promise<undefined>((resolve) => {
          resolve(undefined);
        }),
      afterRead,
    };

    const { input } = createUploader(props);

    fireEvent.change(input(), { target: { files: [mockFile] } });

    expect(afterRead).toHaveBeenCalledTimes(0);
  });

  it('should trigger oversize event when file size is over limit', async () => {
    const onChange = jest.fn();
    const onOversize = jest.fn();

    const props: UploaderProps = {
      maxSize: 1,
      onChange: ([file]) => {
        onChange(file);
      },
      onOversize,
    };

    const { input, rerender } = createUploader(props);

    fireEvent.change(input(), { target: { files: [mockFile] } });

    await waitFor(() => expect(onChange).toBeCalledTimes(0));
    await waitFor(() => expect(onOversize).toBeCalledTimes(1));

    const props2 = {
      ...props,
      multiple: true,
      maxSize: 100000,
    };

    rerender(<Uploader {...props2} />);

    fireEvent.change(input(), { target: { files: [mockFile, mockFile, mockFile] } });
    await waitFor(() => expect(onChange).toBeCalledTimes(1));

    const props3 = {
      ...props,
      multiple: true,
      maxSize: 2,
    };

    rerender(<Uploader {...props3} />);

    fireEvent.change(input(), { target: { files: [mockFile, mockFile, mockFile] } });
    await waitFor(() => expect(onChange).toBeCalledTimes(1));
  });

  it('should trigger clickUpload event when click upload content', async () => {
    const onClickUpload = jest.fn();

    const props: UploaderProps = {
      onClickUpload,
    };

    const { container } = createUploader(props);

    const target = TestsDOM.mustQuerySelector(container, '.rc-uploader__input');

    fireEvent.click(target);

    await waitFor(() => expect(onClickUpload).toBeCalledTimes(1));
  });

  // <Icon name={props.uploadIcon} className={classnames(bem('upload-icon'))} />
  it('should render correctly when icon prop is ReactNode', async () => {
    const props: UploaderProps = {
      uploadIcon: <Icon name="photograph" />,
    };

    const { container, rerender } = createUploader(props);

    expect(container).toMatchSnapshot();

    rerender(<Uploader uploadIcon={null} />);

    expect(container).toMatchSnapshot();
  });

  it('test chooseFile', async () => {
    const Demo = () => {
      const uploaderRef = useRef<UploaderInstance>(null);

      const handleChoose = () => {
        uploaderRef.current?.chooseFile();
      };

      return (
        <div>
          <Uploader ref={uploaderRef} />
          <button type="button" data-testid="button" onClick={handleChoose}>
            chooseFile
          </button>
        </div>
      );
    };

    const { container, getByTestId } = render(<Demo />);

    const buttonBox = getByTestId('button');

    fireEvent.click(buttonBox);

    expect(container).toMatchSnapshot();
  });

  it('test closeImagePreview', async () => {
    const Demo = () => {
      const uploaderRef = useRef<UploaderInstance>(null);

      const handleCloseImagePreview = () => {
        uploaderRef.current?.closeImagePreview();
      };
      const $props = {
        value: [
          {
            isImage: true,
            url: 'https://dummyimage.com/300x200/fff.png',
            status: 'done',
            content: 'https://dummyimage.com/50/fff.png',
          },
        ],
      };

      return (
        <div>
          <Uploader ref={uploaderRef} {...$props} />
          <button type="button" data-testid="button" onClick={handleCloseImagePreview}>
            closeImagePreview
          </button>
        </div>
      );
    };

    const { container, getByTestId } = render(<Demo />);

    const image = TestsDOM.mustQuerySelector(container, '.rc-uploader__preview-image');

    await waitFor(() => {
      fireEvent.click(image);
    });
    const popup: HTMLElement = TestsDOM.mustQuerySelector(document.body, '.rc-image-preview');

    await sleep(600);
    expect(popup).toMatchSnapshot();

    const buttonBox = getByTestId('button');

    await waitFor(() => {
      fireEvent.click(buttonBox);
    });
    await sleep(600);

    expect(popup).toMatchSnapshot();
  });

  it('result-type as text', async () => {
    const onChange = jest.fn();

    const props: UploaderProps = {
      resultType: 'text',
      afterRead(readFile: UploaderFileListItem | UploaderFileListItem[]) {
        expect((readFile as UploaderFileListItem).content).toEqual(mockFileDataUrl);
      },
      onChange: (files) => {
        onChange(files.length);
      },
    };

    const { input } = createUploader(props);

    fireEvent.change(input(), { target: { files: [mockFile] } });

    await waitFor(() => expect(onChange).toBeCalledWith(1));
  });

  it('result-type as file', async () => {
    const onChange = jest.fn();

    const props: UploaderProps = {
      resultType: 'file',
      afterRead(readFile: UploaderFileListItem | UploaderFileListItem[]) {
        expect((readFile as UploaderFileListItem).file).toBeTruthy();
        expect((readFile as UploaderFileListItem).content).toBeFalsy();
      },
      onChange: (files) => {
        onChange(files.length);
      },
    };

    const { input } = createUploader(props);

    fireEvent.change(input(), { target: { files: [mockFile] } });

    await waitFor(() => expect(onChange).toBeCalledWith(1));
  });

  it('should allow to custom max-size for different type of files', async () => {
    const onChange = jest.fn();

    const props: UploaderProps = {
      maxSize(file: File) {
        if (file.type === 'test') {
          return file.size > 500;
        }
        return false;
      },
      onChange: (files) => {
        onChange(files.length);
      },
    };

    const { input } = createUploader(props);

    fireEvent.change(input(), { target: { files: [mockFile] } });

    await waitFor(() => expect(onChange).toBeCalledWith(1));
  });
});
