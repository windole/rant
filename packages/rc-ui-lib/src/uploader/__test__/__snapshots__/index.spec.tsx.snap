// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Uploader/> max-count 1`] = `
<div>
  <div
    class="rc-uploader"
  >
    <div
      class="rc-uploader__wrapper"
    >
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__file"
        >
          <i
            class="rc-badge__wrapper rc-uploader__file-icon van-icon van-icon-description"
          />
          <div
            class="rc-uploader__file-name rc-ellipsis"
          >
            https://upload.com/foo.txt
          </div>
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__preview-image rc-image rc-image--block"
        >
          <img
            alt="img"
            class="rc-image__img"
            src="https://dummyimage.com/50/fff.png"
            style="object-fit: cover;"
          />
          <div
            class="rc-image__loading"
          >
            <i
              class="rc-badge__wrapper rc-image__loading-icon van-icon van-icon-photo"
            />
          </div>
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Uploader/> render children 1`] = `
<div>
  <div
    class="rc-uploader"
  >
    <div
      class="rc-uploader__wrapper"
    >
      <div
        class="rc-uploader__input-wrapper"
      >
        <div>
          children
        </div>
        <input
          accept="image/*"
          class="rc-uploader__input"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Uploader/> should render correctly when icon prop is ReactNode 1`] = `
<div>
  <div
    class="rc-uploader"
  >
    <div
      class="rc-uploader__wrapper"
    >
      <div
        class="rc-uploader__upload"
      >
        <i
          class="rc-badge__wrapper van-icon van-icon-photograph"
        />
        <input
          accept="image/*"
          class="rc-uploader__input"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Uploader/> should render correctly when icon prop is ReactNode 2`] = `
<div>
  <div
    class="rc-uploader"
  >
    <div
      class="rc-uploader__wrapper"
    >
      <div
        class="rc-uploader__upload"
      >
        <input
          accept="image/*"
          class="rc-uploader__input"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Uploader/> test chooseFile 1`] = `
<div>
  <div>
    <div
      class="rc-uploader"
    >
      <div
        class="rc-uploader__wrapper"
      >
        <div
          class="rc-uploader__upload"
        >
          <i
            class="rc-badge__wrapper rc-uploader__upload-icon van-icon van-icon-photograph"
          />
          <input
            accept="image/*"
            class="rc-uploader__input"
            type="file"
          />
        </div>
      </div>
    </div>
    <button
      data-testid="button"
      type="button"
    >
      chooseFile
    </button>
  </div>
</div>
`;

exports[`<Uploader/> test closeImagePreview 1`] = `
<div
  class="rc-popup rc-popup--center rc-image-preview rc-fade-exit-done"
  style="z-index: 2000; animation-duration: 300ms; display: none;"
>
  <div
    class="rc-image-preview__swipe rc-swiper"
    style="--rc-swipe-slide-size: 100%; --rc-swipe-track-offset: 0%;"
  >
    <div
      class="rc-swiper__track rc-swiper__track--allow-touch-move"
    >
      <div
        class="rc-swiper__track-inner"
      >
        <div
          class="rc-swiper__slide"
          style="left: -0%; transform: none;"
        >
          <div
            class="rc-image-preview__item rc-swiper-item"
          >
            <div
              class="rc-image-preview__slide"
              data-testid="slide"
            >
              <div
                class="rc-image-preview__control"
                data-testid="control"
              >
                <div
                  class="rc-image-preview__image-wrapper"
                  style="transform: none;"
                >
                  <div
                    class="rc-image-preview__image rc-image rc-image--block"
                  >
                    <img
                      alt="img"
                      class="rc-image__img"
                      src="https://dummyimage.com/50/fff.png"
                      style="object-fit: contain;"
                    />
                    <div
                      class="rc-image__loading"
                    >
                      <i
                        class="rc-badge__wrapper rc-image__loading-icon van-icon van-icon-photo"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="rc-image-preview__index"
  >
    1 / 1
  </div>
</div>
`;

exports[`<Uploader/> test closeImagePreview 2`] = `
<div
  class="rc-popup rc-popup--center rc-image-preview rc-fade-exit-done"
  style="z-index: 2000; animation-duration: 300ms; display: none;"
>
  <div
    class="rc-image-preview__swipe rc-swiper"
    style="--rc-swipe-slide-size: 100%; --rc-swipe-track-offset: 0%;"
  >
    <div
      class="rc-swiper__track rc-swiper__track--allow-touch-move"
    >
      <div
        class="rc-swiper__track-inner"
      >
        <div
          class="rc-swiper__slide"
          style="left: -0%; transform: none;"
        >
          <div
            class="rc-image-preview__item rc-swiper-item"
          >
            <div
              class="rc-image-preview__slide"
              data-testid="slide"
            >
              <div
                class="rc-image-preview__control"
                data-testid="control"
              >
                <div
                  class="rc-image-preview__image-wrapper"
                  style="transform: none;"
                >
                  <div
                    class="rc-image-preview__image rc-image rc-image--block"
                  >
                    <img
                      alt="img"
                      class="rc-image__img"
                      src="https://dummyimage.com/50/fff.png"
                      style="object-fit: contain;"
                    />
                    <div
                      class="rc-image__loading"
                    >
                      <i
                        class="rc-badge__wrapper rc-image__loading-icon van-icon van-icon-photo"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="rc-image-preview__index"
  >
    1 / 1
  </div>
</div>
`;

exports[`<Uploader/> test previewCoverRender 1`] = `
<div>
  <div
    class="rc-uploader"
  >
    <div
      class="rc-uploader__wrapper"
    >
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__file"
        >
          <i
            class="rc-badge__wrapper rc-uploader__file-icon van-icon van-icon-description"
          />
          <div
            class="rc-uploader__file-name rc-ellipsis"
          >
            https://upload.com/foo.txt
          </div>
          <div
            class="rc-uploader__preview-cover"
          >
            <p>
              https://upload.com/foo.txt
            </p>
          </div>
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__preview-image rc-image rc-image--block"
        >
          <img
            alt="img"
            class="rc-image__img"
            src="https://dummyimage.com/50/fff.png"
            style="object-fit: cover;"
          />
          <div
            class="rc-image__loading"
          >
            <i
              class="rc-badge__wrapper rc-image__loading-icon van-icon van-icon-photo"
            />
          </div>
          <div
            class="rc-uploader__preview-cover"
          >
            <p>
              https://dummyimage.com/300x200/fff.png
            </p>
          </div>
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__upload"
      >
        <i
          class="rc-badge__wrapper rc-uploader__upload-icon van-icon van-icon-photograph"
        />
        <input
          accept="image/*"
          class="rc-uploader__input"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Uploader/> test remove 1`] = `
<div>
  <div
    class="rc-uploader"
  >
    <div
      class="rc-uploader__wrapper"
    >
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__file"
        >
          <i
            class="rc-badge__wrapper rc-uploader__file-icon van-icon van-icon-description"
          />
          <div
            class="rc-uploader__file-name rc-ellipsis"
          >
            https://upload.com/foo.txt
          </div>
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__preview-image rc-image rc-image--block"
        >
          <img
            alt="img"
            class="rc-image__img"
            src="https://dummyimage.com/50/fff.png"
            style="object-fit: cover;"
          />
          <div
            class="rc-image__loading"
          >
            <i
              class="rc-badge__wrapper rc-image__loading-icon van-icon van-icon-photo"
            />
          </div>
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__upload"
      >
        <i
          class="rc-badge__wrapper rc-uploader__upload-icon van-icon van-icon-photograph"
        />
        <input
          accept="image/*"
          class="rc-uploader__input"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Uploader/> test remove 2`] = `
<div>
  <div
    class="rc-uploader"
  >
    <div
      class="rc-uploader__wrapper"
    >
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__file"
        >
          <i
            class="rc-badge__wrapper rc-uploader__file-icon van-icon van-icon-description"
          />
          <div
            class="rc-uploader__file-name rc-ellipsis"
          >
            https://upload.com/foo.txt
          </div>
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__preview-image rc-image rc-image--block"
        >
          <img
            alt="img"
            class="rc-image__img"
            src="https://dummyimage.com/50/fff.png"
            style="object-fit: cover;"
          />
          <div
            class="rc-image__loading"
          >
            <i
              class="rc-badge__wrapper rc-image__loading-icon van-icon van-icon-photo"
            />
          </div>
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__upload"
      >
        <i
          class="rc-badge__wrapper rc-uploader__upload-icon van-icon van-icon-photograph"
        />
        <input
          accept="image/*"
          class="rc-uploader__input"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Uploader/> thumbnail 1`] = `
<div>
  <div
    class="rc-uploader"
  >
    <div
      class="rc-uploader__wrapper"
    >
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__file"
        >
          <i
            class="rc-badge__wrapper rc-uploader__file-icon van-icon van-icon-description"
          />
          <div
            class="rc-uploader__file-name rc-ellipsis"
          />
        </div>
        <div
          class="rc-uploader__mask"
        >
          <div
            class="rc-uploader__loading rc-loading rc-loading--circular"
          >
            <span
              class="rc-loading__spinner rc-loading__spinner--circular"
            >
              <svg
                class="rc-loading__circular"
                viewBox="25 25 50 50"
              >
                <circle
                  cx="50"
                  cy="50"
                  fill="none"
                  r="20"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__file"
        >
          <i
            class="rc-badge__wrapper rc-uploader__file-icon van-icon van-icon-description"
          />
          <div
            class="rc-uploader__file-name rc-ellipsis"
          />
        </div>
        <div
          class="rc-uploader__mask"
        >
          <i
            class="rc-badge__wrapper rc-uploader__mask-icon van-icon van-icon-close"
          />
          <div
            class="rc-uploader__mask-message"
          >
            failed
          </div>
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__file"
        >
          <i
            class="rc-badge__wrapper rc-uploader__file-icon van-icon van-icon-description"
          />
          <div
            class="rc-uploader__file-name rc-ellipsis"
          />
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__preview"
      >
        <div
          class="rc-uploader__file"
        >
          <i
            class="rc-badge__wrapper rc-uploader__file-icon van-icon van-icon-description"
          />
          <div
            class="rc-uploader__file-name rc-ellipsis"
          />
        </div>
        <div
          class="rc-uploader__preview-delete"
        >
          <i
            class="rc-badge__wrapper rc-uploader__preview-delete-icon van-icon van-icon-cross"
          />
        </div>
      </div>
      <div
        class="rc-uploader__upload"
      >
        <i
          class="rc-badge__wrapper rc-uploader__upload-icon van-icon van-icon-photograph"
        />
        <input
          accept="image/*"
          class="rc-uploader__input"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;
